### PR TITLE
Changing subroute behavior

### PR DESCRIPTION
**Pull request by luchotc**
Solves #37

### CODE REVIEW

File: lib/mumukit/platform/uri.rb [hunk 1]
```diff
@@ -8,11 +8,11 @@ def subdominate(subdomain)
     rebuild(host: new_host)
   end
 
-  def subroute(route)
-    if path.start_with? '/'
-      new_path = "/#{route}#{path}"
+  def tenantize(route)
```
Comment (luchotc) [inline lib/mumukit/platform/uri.rb:11]: I'm not happy with the name :confused:

File: spec/mumukit/uri_spec.rb [hunk 2]
```diff
@@ -19,13 +19,12 @@ def add_extra_path(original_url)
     it { expect(add_extra_subdomain('http://foo.bar.com/foo?z=3')).to eq 'http://extra.foo.bar.com/foo?z=3' }
     it { expect(add_extra_subdomain('http://foo.bar.com:3000/foo?z=3')).to eq 'http://extra.foo.bar.com:3000/foo?z=3' }
   end
-  describe '#subroute' do
+  describe '#tenantize' do
     it { expect(add_extra_path('https://foo.bar')).to eq 'https://foo.bar/extra/' }
     it { expect(add_extra_path('http://foo.bar')).to eq 'http://foo.bar/extra/' }
-    it { expect(add_extra_path('http://foo.bar/zaraza')).to eq 'http://foo.bar/extra/zaraza' }
+    it { expect(add_extra_path('http://foo.bar/zaraza')).to eq 'http://foo.bar/zaraza/extra/' }
```
Comment (luchotc) [inline spec/mumukit/uri_spec.rb:25]: This is the main change here. See more info in linked issue

File: spec/mumukit/uri_spec.rb [hunk 3]
```diff
@@ -19,13 +19,12 @@ def add_extra_path(original_url)
...
 
     it { expect(add_extra_path('http://foo.bar.com/')).to eq 'http://foo.bar.com/extra/' }
-    it { expect(add_extra_path('http://foo.bar.com/foo?z=3')).to eq 'http://foo.bar.com/extra/foo?z=3' }
-    it { expect(add_extra_path('http://foo.bar.com:3000/foo?z=3')).to eq 'http://foo.bar.com:3000/extra/foo?z=3' }
```
Comment (luchotc) [inline spec/mumukit/uri_spec.rb:29]: I'm not testing this nor supporting it as this method is used only for adding a tenant for base_url.

Comment (flbulgarelli) [reply spec/mumukit/uri_spec.rb:29]: :thinking: I am wondering that may introduce subtle issues in the future

File: lib/mumukit/platform/uri.rb (hunk 1 above)
Comment (flbulgarelli) [reply lib/mumukit/platform/uri.rb:11]: :thought_balloon: From my POV `tenantize` is just another case of "subrouting". Perhaps `subroute(route, append: true)` 

Another option my by just calling it `append`: `uri.append(route)`. However with that name I'd strongly suggest handling query params and fragments.

**Review by flbulgarelli — COMMENTED** [review]
See my comments.

File: spec/mumukit/uri_spec.rb (hunk 3 above)
Comment (flbulgarelli) [reply spec/mumukit/uri_spec.rb:29]: Perhaps if we don't support that, we should raise an exception. We have had already some bugs that we causing that improper url building based on implicit preconditions that weren't match.

**Comment by julian-berbel** [comment]
After talking a bit about it we're evaluating moving the app prepending logic to Application, that way, an url would be something like `"#{application_namespace}/#{organic_url(uri)}"`

File: spec/mumukit/uri_spec.rb (hunk 3 above)
Comment (julian-berbel) [reply spec/mumukit/uri_spec.rb:29]: Well, we just tested this and although a '/' is appended to the url, url params still work! so I'm just readding those tests

File: lib/mumukit/platform/uri.rb (hunk 1 above)
Comment (julian-berbel) [reply lib/mumukit/platform/uri.rb:11]: We've discussed this and are happy with the name after all! :laughing:

**Pull request merged**